step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/execution/IsolatedAutomationPackageRepository.java

```diff
@@ -177,6 +181,55 @@ public void cleanUpOutdatedResources() {
         log.info("Cleanup outdated automation packages finished. {} of {} packages have been removed", removed, foundResources.size());
     }
 
+    public void cleanUpMavenCache() {
+        log.info("Cleanup outdated artifacts in maven cache...");
+        String ttlString = ttlValueSupplier.get();
+
+        long ttlDurationMs = Long.parseLong(ttlString);
+        Long minExecutionTime = System.currentTimeMillis() - ttlDurationMs;
+
+        int removed = 0;
+        int artifactsCandidate = 0;
+        try {
+            Path mavenCachePath = Path.of("maven");
+            List<Path> mavenJarFiles = findOldJarFiles(mavenCachePath, minExecutionTime);
+            artifactsCandidate = mavenJarFiles.size();
+            for(Path jarPath: mavenJarFiles) {
+                File parentFile = jarPath.toFile().getParentFile();
+                if (parentFile != null) {
+                    boolean deleted = FileHelper.safeDeleteFolder(parentFile);;
```

Comment:
Theoretically we should acquire a write lock shared with the maven repository plugin in order to ensure that the file will not be read at the same time. The probability is however very low

Reply:
Agreed, we should probably do it in a next iterations but also for the cleanup of the AP cache